editor.planx.uk/src/@planx/components/TextInput/Public.tsx

```diff
@@ -62,13 +69,22 @@ const TextInputComponent: React.FC<Props> = (props) => {
             id={props.id}
             inputProps={{
               "aria-describedby": [
-                props.description ? DESCRIPTION_TEXT : "",
+                props.description
+                  ? `${DESCRIPTION_TEXT} character-hint`
+                  : "character-hint",
                 formik.errors.text ? `${ERROR_MESSAGE}-${props.id}` : "",
               ]
                 .filter(Boolean)
                 .join(" "),
             }}
           />
+          {characterCountLimit && (
+            <CharacterCounter
+              count={formik.values.text.length}
+              limit={characterCountLimit}
+              error={formik.errors.text ? true : false}
```

Comment:
nit: Maybe a bit cleaner?

```suggestion
              error={Boolean(formik.errors.text)}
```